### DISCUSSION TITLE
Shell task Run jar with heap size

### DISCUSSION THREAD

**Discussion by cumtqiangqiang** [apache/dolphinscheduler#10192] [Q&A]
In shell task ,we will run jar with heap size 20G (For example), and another same  task will also use 20G, but our server is only 30G, is there some config to let the second task to wait until the first completed.

**Comment by cumtqiangqiang** [comment]
@Tandoy @lenboo  Looking forward to your reply，thx

**Reply by Tandoy** [reply]
> @Tandoy @lenboo Looking forward to your reply，thx

If concurrent execution is not required,maybe you can increase the order of task execution and duild DAG

https://dolphinscheduler.apache.org/en-us/docs/1.3.6/user_doc/system-manual.html

![企业微信截图_20220523173235](https://user-images.githubusercontent.com/56899730/169789906-e1083ca2-ec5f-4a0c-9530-496b7ddea15d.png)

**Reply by cumtqiangqiang** [reply]
> > @Tandoy @lenboo Looking forward to your reply，thx
> 
> If concurrent execution is not required,maybe you can increase the order of task execution and duild DAG
> 
> https://dolphinscheduler.apache.org/en-us/docs/1.3.6/user_doc/system-manual.html
> 
> ![企业微信截图_20220523173235](https://user-images.githubusercontent.com/56899730/169789906-e1083ca2-ec5f-4a0c-9530-496b7ddea15d.png)

Thanks for your replay, we have many similar jobs, sometimes need to backfill use a temp dag,so concurrent execution is required sometime,  using the order ,it is the optimal solution at present right ? @Tandoy

**Reply by Tandoy** [reply]
> > > @Tandoy @lenboo Looking forward to your reply，thx
> > 
> > 
> > If concurrent execution is not required,maybe you can increase the order of task execution and duild DAG
> > https://dolphinscheduler.apache.org/en-us/docs/1.3.6/user_doc/system-manual.html
> > ![企业微信截图_20220523173235](https://user-images.githubusercontent.com/56899730/169789906-e1083ca2-ec5f-4a0c-9530-496b7ddea15d.png)
> 
> Thanks for your replay, we have many similar jobs, sometimes need to backfill use a temp dag,so concurrent execution is required sometime, using the order ,it is the optimal solution at present right ? @Tandoy

The temporary repair can take the form of appeal.
Subsequent optimization scheme: you can set the system resource threshold and automatically select the parallel or serial running mode in cooperation with the conditions task node before starting the task.

https://dolphinscheduler.apache.org/zh-cn/docs/latest/user_doc/guide/task/conditions.html

**Comment by cumtqiangqiang** [comment]
When I run a dag and use the complement process, complement process has reached resource limits, there are two two scenarios：
1. when other dag submit a task to this sever ,then this task will wait.
2. the complement process has not done yet,the scheduling task start to run ,however this scheduling task doesnot wait,then  complement process will be killed because of there resource limit.

I think the first scenario is that we need, the second is normal?  @Tandoy @lenboo Looking forward to your reply，thx

**Reply by Tandoy** [reply]
> When I run a dag and use the complement process, complement process has reached resource limits, there are two two scenarios：
> 
> 1. when other dag submit a task to this sever ,then this task will wait.
> 2. the complement process has not done yet,the scheduling task start to run ,however this scheduling task doesnot wait,then  complement process will be killed because of there resource limit.
> 
> I think the first scenario is that we need, the second is normal? @Tandoy @lenboo Looking forward to your reply，thx

yes, the second is normal and will be sent.If you want to achieve the first situation, you can use the Condition node to judge the system resources before normal scheduling or complement data.

**Reply by cumtqiangqiang** [reply]
> > When I run a dag and use the complement process, complement process has reached resource limits, there are two two scenarios：
> > 
> > 1. when other dag submit a task to this sever ,then this task will wait.
> > 2. the complement process has not done yet,the scheduling task start to run ,however this scheduling task doesnot wait,then  complement process will be killed because of there resource limit.
> > 
> > I think the first scenario is that we need, the second is normal? @Tandoy @lenboo Looking forward to your reply，thx
> 
> yes, the second is normal and will be sent.If you want to achieve the first situation, you can use the Condition node to judge the system resources before normal scheduling or complement data.
![question](https://user-images.githubusercontent.com/14304089/171631417-6a2f1720-1722-42d3-affb-e98d2364670e.jpg)

Thanks for your kind reply.There is an issue I am a little confused.For dag1  I need complement and schedule,How can I use a condition task to check the system resources before normal scheduling or complement data. According the doc, condition task based on the previous task, I think it is just run once.Do I understand this correctly ?

Happy Dragon Boat Festival. @Tandoy

**Comment by Tandoy** [comment]
you can set the failure retry of the Python node or the Shell node to achieve multiple judgments,like this DAG.  8D

![Snipaste_2022-06-03_09-53-47](https://user-images.githubusercontent.com/56899730/171773483-6c12edfb-e395-4b48-87da-b01ca55c3398.PNG)

**Reply by cumtqiangqiang** [reply]
> you can set the failure retry of the Python node or the Shell node to achieve multiple judgments,like this DAG. 8D
> 
> ![Snipaste_2022-06-03_09-53-47](https://user-images.githubusercontent.com/56899730/171773483-6c12edfb-e395-4b48-87da-b01ca55c3398.PNG)

Thanks for this advice， I will try it.

**Reply by cumtqiangqiang** [reply]
@Tandoy  Could you please invite me to Slack Group,Thanks.

**Reply by Tandoy** [reply]
> @Tandoy Could you please invite me to Slack Group,Thanks.

Please give me your email address

**Reply by cumtqiangqiang** [reply]
> > @Tandoy Could you please invite me to Slack Group,Thanks.
> 
> Please give me your email address

cumtqiangqiang@163.com @Tandoy  thx

**Reply by Tandoy** [reply]
> cumtqiangqiang@163.com

workspace has exceeded the limit for invitations. Maybe you can try the following link
https://join.slack.com/t/asf-dolphinscheduler/shared_invite/zt-1bzca1dsl-A7pCLBEWTEedkFey1teCCw
